src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/cas2/StatusUpdateServiceTest.kt

```diff
@@ -429,4 +445,309 @@ class StatusUpdateServiceTest {
       }
     }
   }
+
+  @Nested
+  inner class CreateForAssessment {
+
+    @Nested
+    inner class WhenSuccessful {
+
+      @BeforeEach
+      fun setUp() {
+        val cas2StatusUpdateEntity = Cas2StatusUpdateEntityFactory()
+          .withApplication(application)
+          .withAssessor(assessor)
+          .withStatusId(activeStatus.id)
+          .produce()
+
+        every { mockAssessmentRepository.findByIdOrNull(assessment.id) } answers
+          {
+            assessment
+          }
+
+        every { mockStatusUpdateRepository.save(any()) } answers
+          {
+            cas2StatusUpdateEntity
+          }
+
+        every { mockDomainEventService.saveCas2ApplicationStatusUpdatedDomainEvent(any()) } answers { }
+
+        every { mockNotifyConfig.templates.cas2ApplicationStatusUpdated } returns "abc123"
+
+        every {
+          mockEmailNotificationService.sendCas2Email(
+            recipientEmailAddress = applicant.email!!,
+            templateId = "abc123",
+            personalisation = mapOf(
+              "applicationStatus" to cas2StatusUpdateEntity.label,
+              "dateStatusChanged" to cas2StatusUpdateEntity.createdAt.toLocalDate().toCas2UiFormat(),
+              "timeStatusChanged" to cas2StatusUpdateEntity.createdAt.toCas2UiFormattedHourOfDay(),
+              "nomsNumber" to "NOMSABC",
+              "applicationType" to "Home Detention Curfew (HDC)",
+              "applicationUrl" to "http://example.com/application/$applicationId/overview",
+            ),
+          )
+        } just Runs
+      }
+
+      @Test
+      fun `saves and asks the domain event service to create a status-updated event`() {
+        statusUpdateService.createForAssessment(
+          assessmentId = assessment.id,
+          statusUpdate = applicationStatusUpdate,
+          assessor = assessor,
+        )
+
+        verify {
+          mockDomainEventService.saveCas2ApplicationStatusUpdatedDomainEvent(
+            match {
+              it.crn == "CRN123" &&
+                it.applicationId == applicationId &&
+                it.data.eventType == EventType.applicationStatusUpdated &&
+                it.data.eventDetails.applicationId == applicationId &&
```

Comment:
AssessmentId check?

Reply:
I don't think we add the assessment to the domain event?